server/service/src/report/report_service.rs

```diff
@@ -568,11 +579,35 @@ fn transform_data(
     connection: StorageConnection,
     data: ReportData,
     convert_data: Option<String>,
-) -> Result<ReportData, ReportError> {
+    convert_data_type: &ConvertDataType,
+) -> Result<ReportData, ConvertDataError> {
     let Some(convert_data) = convert_data else {
         return Ok(data);
     };
 
+    match convert_data_type {
+        // Mapping to string via format_error since it's better then debug output on error
+        ConvertDataType::BoaJs => transform_data_boajs(data, convert_data)
+            .map_err(|e| ConvertDataError::BoaJs(format_error(&e))),
+        ConvertDataType::Extism => {
+            transform_data_extism(connection, data, convert_data).map_err(ConvertDataError::Extism)
+        }
+    }
+}
+
+fn transform_data_boajs(data: ReportData, convert_data: String) -> Result<ReportData, BoaJsError> {
+    call_method(
+        data,
+        vec!["convert_data"],
+        &BASE64_STANDARD.decode(convert_data).unwrap(),
```

Comment:
could do base64 decode in transform_data, but I didn't because of variable naming, if it was called convert_data_base64 then yeah would have made more sense, on hindsight should have been specific (but thought that we could store it as none base64 in the future ¯\_(ツ)_/¯l